Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/CarouselViewGalleries/EmptyCarouselGallery.xaml.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Windows.Input;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms.Controls.GalleryPages.CollectionViewGalleries.CarouselViewGalleries
+{
+	[Preserve(AllMembers = true)]
+	public partial class EmptyCarouselGallery : ContentPage
+	{
+		public EmptyCarouselGallery()
+		{
+			InitializeComponent();
+			BindingContext = new EmptyCarouselGalleryViewModel();
+		}
+	}
+
+	[Preserve(AllMembers = true)]
+	public class EmptyCarouselGalleryViewModel : BindableObject
+	{
+		ObservableCollection<CarouselData> _items;
+
+		public EmptyCarouselGalleryViewModel()
+		{
+			Items = new ObservableCollection<CarouselData>();
+		}
+
+		public ObservableCollection<CarouselData> Items
+		{
+			get { return _items; }
+			set
+			{
+				_items = value;
+				OnPropertyChanged();
+			}
+		}
+
+		public ICommand AddCommand => new Command(Add);
+
+		public ICommand ClearCommand => new Command(Clear);
+
+		void LoadItems()
+		{
+			var random = new Random();
+
+			if (Device.RuntimePlatform == Device.iOS)
```

Comment:
This code (different by platforms) is due to a different issue: https://github.com/xamarin/Xamarin.Forms/issues/7678